lib/connectors_async/job_runner.rb

```diff
@@ -37,16 +37,15 @@ def start_job(job:, connector_class:, secret_storage:, params:)
         job.update_status(ConnectorsShared::JobStatus::FINISHED)
         job.update_cursors(new_cursors)
 
-        log("Job #{job.id} has finished successfully")
+        log_with_thread_id("Job #{job.id} has finished successfully")
       rescue StandardError => e
         job.fail(e)
-        log("Job #{job.id} failed: #{e.message}")
+        log_with_thread_id("Job #{job.id} failed: #{e.message}")
       end
     end
 
-    def log(str)
-      # TODO: use proper logging
-      puts("[#{Time.now.to_i}] [Thread #{Thread.current.object_id}] #{str}")
+    def log_with_thread_id(str)
+      ConnectorsShared::Logger.debug("[#{Time.now.to_i}] [Thread #{Thread.current.object_id}] #{str}")
```

Comment:
Do you think we should let the caller specify the log level? The first 2 calls are more appropriate to `info` and apparently, the last one is `error`

Reply:
Good call, I changed the implementation accordingly